### PR TITLE
refactor(IMS): fix ims data source lint issues

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  fix ims data source lint issues
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  fix ims data source lint issues
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ims/data_source_huaweicloud_images_image_test.go'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims/data_source_huaweicloud_images_image_test.go -v  -timeout 360m -parallel 4
=== RUN   TestAccImsImageDataSource_basic
=== PAUSE TestAccImsImageDataSource_basic
=== RUN   TestAccImsImageDataSource_testQueries
=== PAUSE TestAccImsImageDataSource_testQueries
=== CONT  TestAccImsImageDataSource_basic
=== CONT  TestAccImsImageDataSource_testQueries
--- PASS: TestAccImsImageDataSource_basic (53.42s)
--- PASS: TestAccImsImageDataSource_testQueries (440.91s) 
PASS
ok      command-line-arguments  440.948s

make testacc TEST='./huaweicloud/services/acceptance/ims/data_source_huaweicloud_images_images_test.go'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims/data_source_huaweicloud_images_images_test.go -v  -timeout 360m -parallel 4
=== RUN   TestAccImsImagesDataSource_basic
=== PAUSE TestAccImsImagesDataSource_basic
=== RUN   TestAccImsImagesDataSource_testQueries
=== PAUSE TestAccImsImagesDataSource_testQueries
=== CONT  TestAccImsImagesDataSource_basic
=== CONT  TestAccImsImagesDataSource_testQueries
--- PASS: TestAccImsImagesDataSource_basic (57.22s)
--- PASS: TestAccImsImagesDataSource_testQueries (431.35s) 
PASS
ok      command-line-arguments  431.392s
```

### CODE REVIEW

File: huaweicloud/services/ims/data_source_huaweicloud_images_image.go [hunk 1]
```diff
@@ -47,14 +39,12 @@ func DataSourceImagesImageV2() *schema.Resource {
 				Type:          schema.TypeString,
 				Optional:      true,
 				ConflictsWith: []string{"name"},
-				ValidateFunc:  validation.StringIsValidRegExp,
```
Comment (ShiChangkuo) [inline huaweicloud/services/ims/data_source_huaweicloud_images_image.go]: we can keep the validation for regex.

File: huaweicloud/services/ims/data_source_huaweicloud_images_image.go [hunk 2]
```diff
@@ -221,27 +206,27 @@ func dataSourceImagesImageV2Read(ctx context.Context, d *schema.ResourceData, me
...
+	if epsId := common.GetEnterpriseProjectID(d, cfg); epsId != "" {
 		listOpts.EnterpriseProjectID = epsId
 	} else {
 		listOpts.EnterpriseProjectID = "all_granted_eps"
 	}
 
-	logp.Printf("[DEBUG] List Options: %#v", listOpts)
+	log.Printf("[DEBUG] List Options: %#v", listOpts)
 
 	var image cloudimages.Image
```
Comment (ShiChangkuo) [inline huaweicloud/services/ims/data_source_huaweicloud_images_image.go]: I think we can define it at L254

File: huaweicloud/services/ims/data_source_huaweicloud_images_image.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/ims/data_source_huaweicloud_images_image.go]: done

File: huaweicloud/services/ims/data_source_huaweicloud_images_image.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/ims/data_source_huaweicloud_images_image.go]: done

**Pull request merged**